charts/hlf-k8s/templates/deployment-appchannel-operator.yaml

```diff
@@ -210,20 +222,67 @@ spec:
 
               done < /config/application-organizations
 
-              ## Instantiate chaincode
-              peer chaincode -C {{ .channelName }} list --instantiated > chaincode.list
 
-              until grep "{{ .chaincodeName }}" chaincode.list | grep "{{ .chaincodeVersion }}" > /dev/null; do
+              ## CHAINCODES
 
-                printf "[DEBUG] Instantiate chaincode {{ .chaincodeName }} {{ .chaincodeVersion }} on channel {{ .channelName }} with policy : {{ .chaincodePolicy }}\n"
-                peer chaincode instantiate -C {{ .channelName }} -n {{ .chaincodeName }} -v {{ .chaincodeVersion }} -c '{"Args":["init"]}' -P "{{ .chaincodePolicy }}" -o {{ index $.Values "hlf-ord" "host" }}:{{ index $.Values "hlf-ord" "port" }} --tls --clientauth --cafile /var/hyperledger/tls/ord/cert/cacert.pem --keyfile /var/hyperledger/tls/client/pair/tls.key --certfile /var/hyperledger/tls/client/pair/tls.crt
-                sleep 5
+              {{- range .chaincodes }}
+
+              ## Wait readiness
+
+              while true; do
 
-                peer chaincode -C {{ .channelName }} list --instantiated > chaincode.list
+                # Chaincode commit
+                printf "[DEBUG] Wait chaincode ready {{ .chaincodeName }} {{ .chaincodeVersion }} on channel {{ $value.channelName }}\n"
+
+                peer lifecycle chaincode checkcommitreadiness --signature-policy "{{ .chaincodePolicy}}" --channelID {{ $value.channelName }} --name {{ .chaincodeName }} --version {{ .chaincodeVersion }} --sequence 1 --init-required -o {{ index $.Values "hlf-ord" "host" }}:{{ index $.Values "hlf-ord" "port" }} --tls --clientauth --cafile /var/hyperledger/tls/ord/cert/cacert.pem --keyfile /var/hyperledger/tls/client/pair/tls.key --certfile /var/hyperledger/tls/client/pair/tls.crt > chaincode.{{ .chaincodeName }}.ready 2>&1
```

Comment:
For maintenance and readability I would rely on environment variables for all those options: https://hyperledger-fabric.readthedocs.io/en/release-2.2/enable_tls.html#configuring-tls-for-peers-nodes (this is for example the tls variables to set)

Reply:
Against an orderer, we cannot use environment variales : https://hyperledger-fabric.readthedocs.io/en/release-2.2/enable_tls.html#configuring-tls-for-the-peer-cli

```
When running a command that connects to orderer service, like peer channel <create|update|fetch> 
or peer chaincode <invoke>, following command line arguments must also be specified if TLS is enabled 
on the orderer:

–tls
–cafile <fully qualified path of the file that contains cert chain of the orderer CA>
If TLS client authentication is enabled on the orderer, the following arguments must be specified as well:

–clientauth
–keyfile <fully qualified path of the file that contains the client private key>
–certfile <fully qualified path of the file that contains the client certificate>
```